mysql-test/main/auto_increment.test

```diff
@@ -433,3 +439,28 @@ SET @engine='MyISAM';
 --echo # End of 10.2 tests
 --echo #
 
+--echo #
+--echo # Start of 10.4 tests
+--echo #
+
+--echo #
+--echo # MDEV-32018 Allow the setting of Auto_increment on FK referenced columns
+--echo #
+
+CREATE TABLE t1 (
+  id int unsigned NOT NULL PRIMARY KEY
+);
+
+CREATE TABLE t2 (
+  id int unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  t1_id int unsigned DEFAULT NULL,
+  CONSTRAINT FK_t1_id FOREIGN KEY (t1_id) REFERENCES t1 (id)
+);
+
+ALTER TABLE t1 MODIFY id INT unsigned AUTO_INCREMENT;
```

Comment:
This test does not appear to enable InnoDB. Therefore, the foreign key constraint should be ignored by whatever is the default storage engine.

I would suggest to add a test that uses explicit `ALGORITHM=INPLACE` so that the error message that was added to `ha_innobase::check_if_supported_inplace_alter()` will be covered.

Reply:
wasn't reachable so left as commit.